### DISCUSSION TITLE
Is it possible to give an action's GITHUB_TOKEN read access to all Org/Enterprise's repositories?

### DISCUSSION THREAD

**Discussion by wspurgin** [community/community#38232] [Enterprise]
I'm building a GitHub Action to automate dependency updates for a Ruby app. Under the hood, this action runs a Ruby script that uses [Octokit](https://github.com/octokit/octokit.rb) to pull information (Release notes, changelog, etc. etc.) about any updated dependencies that have their code up on GitHub.

This works great for any public dependencies, _but_ for my Org's private/internal gems, the GitHub API (through Octokit) returns 404s for the gems' repos (here on GitHub) 😭 

Is it even possible to give an Action's GITHUB_TOKEN Org level read access to internal/private repos for the Org? Or is my only option a PAT?

-- edit --

Just to be clear, I have spelunked around and I've seen interesting developments for granting actions read permissions to Org/Enterprise _pacakges_ (https://github.com/github/roadmap/issues/554 and [more coming](https://github.com/actions/setup-node/issues/49#issuecomment-1249822367)), just nothing to indicate that you can give an Action's GITHUB_TOKEN access to read any other repos that isn't the one referencing the action.

**Comment by ossanna16** [comment]
Hi there @wspurgin and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)
